core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -525,14 +523,43 @@ private void syncRemoteChangeEventToLocal(
     final BsonDocument docFilter = getDocumentIdFilter(docConfig.getDocumentId());
     final BsonDocument localDocument = localColl.find(docFilter).first();
 
-    final BsonValue lastKnownRemoteVersion;
+    final BsonDocument lastKnownRemoteVersion;
     if (remoteDoc == null || remoteDoc.size() == 0) {
       lastKnownRemoteVersion = null;
     } else {
-      lastKnownRemoteVersion = DocumentVersionInfo
-          .getRemoteVersionInfo(remoteDoc).getCurrentVersion();
+      final DocumentVersionInfo remoteVersionInfo = DocumentVersionInfo
+              .getRemoteVersionInfo(remoteDoc);
+      lastKnownRemoteVersion = remoteVersionInfo.getVersionDoc();
+
+      final DocumentVersionInfo localVersion = DocumentVersionInfo.getLocalVersionInfo(docConfig);
+
+      if (remoteVersionInfo.isNonEmptyVersion()
+          && remoteVersionInfo.getSyncProtocolVersion() != 1) {
+        desyncDocumentFromRemote(nsConfig.getNamespace(), docConfig.getDocumentId());
+
+        emitError(docConfig,
+                String.format(
+                        Locale.US,
+                        "t='%d': syncRemoteChangeEventToLocal ns=%s documentId=%s got a remote "
+                                + "document with an unsupported synchronization protocol version "
+                                + "%d; dropping the event, and desyncing the document",
+                        logicalT,
+                        nsConfig.getNamespace(),
+                        docConfig.getDocumentId(),
+                        remoteVersionInfo.getSyncProtocolVersion()));
 
-      if (docConfig.hasCommittedVersion(lastKnownRemoteVersion)) {
+        return;
+      }
+
+      // TODO(QUESTION FOR REVIEW): The testFrozenDocumentConfig test fails if I call this, but
```

Comment:
There reason not to set this and why you can get rid of that set method is this would always make the next statement true.